consensus/common/src/main/java/tech/pegasys/pantheon/consensus/common/VoteProposer.java

```diff
@@ -128,7 +127,9 @@ private boolean voteNotYetCast(
     // Get the next position in the voting queue we should propose
     final int currentVotePosition = votePosition.updateAndGet(i -> ++i % validVotes.size());
 
+    final Map.Entry<Address, VoteType> voteToCast = validVotes.get(currentVotePosition);
+
     // Get a vote from the valid votes and return it
-    return Optional.of(validVotes.get(currentVotePosition));
+    return Optional.of(new ValidatorVote(voteToCast.getValue(), localAddress, voteToCast.getKey()));
```

Comment:
Outside the scope of this PR, but using `Map.Entry` so much here feels like a bit of an abuse. It works but it's an exceptionally generic type to use for what really should be a business domain concept - I suspect `validVotes` should  be `List<ValidatorVote>` and that may even flow up to `proposals` being `Map<Address, ValidatorVote>` if localAddress was available when calling `auth` and `drop`.

Reply:
+1 to returning `Optional<ValidatorVote>` from this method instead of `Optional<Map.Entry<Address, VoteType>>` btw.